### ISSUE TITLE
Can't start local devnet with matic-cli 

### ISSUE THREAD

**Issue by gladcow** [maticnetwork/matic-cli#152]
I can't start local devnet on Ubuntu 20 (I've tried several hosts including fresh installation). I'm using `v0.0.9` tag and default config file.
My exact actions:
```
sudo apt update
sudo apt install build-essential
wget https://raw.githubusercontent.com/maticnetwork/node-ansible/master/go-install.sh
bash go-install.sh --remove
bash go-install.sh
sudo apt install rabbitmq-server
curl -fsSL https://get.docker.com -o get-docker.sh
sudo sh get-docker.sh
sudo usermod -aG docker $USER
newgrp docker
curl https://raw.githubusercontent.com/creationix/nvm/master/install.sh | bash
nvm install 16.17.1
sudo apt install nodejs npm
sudo apt install python2 && alias python="/usr/bin/python2
sudo snap install solc
sudo npm install -g ganache
git clone https://github.com/maticnetwork/matic-cli.git
cd matic-cli
git checkout v0.0.9
npm i
mkdir devnet
cd devnet
../bin/matic-cli setup devnet -c ../configs/devnet/docker-setup-config.yaml
```
The error I'm getting is:
```
  ✔ Clone Heimdall repository
  ✔ Build Heimdall
  ✔ Create testnet files for Heimdall
  ✔ Setup accounts
  ✔ Setup genesis contracts
  ✔ Setup Bor keystore and genesis files
  ❯ Setup contracts on Ganache
    ✔ Clone matic contracts repository
    ✔ Install dependencies for matic contracts
    ✔ Process templates
    ✖ Compile matic contracts
      → Command failed: npm run truffle:compile
      Process scripts
      Deploy contracts
      Prepare contract addresses
      Load contract addresses
    Remove multiple keystore files
    Docker
/home/jahr/ankr/matic-cli/node_modules/execa/index.js:174
		err = new Error(`Command failed: ${joinedCmd}${output}`);
		      ^

Error: Command failed: npm run truffle:compile
    at makeError (/home/jahr/ankr/matic-cli/node_modules/execa/index.js:174:9)
    at /home/jahr/ankr/matic-cli/node_modules/execa/index.js:278:16
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5) {
  code: 1,
  stdout: null,
  stderr: null,
  failed: true,
  signal: null,
  cmd: 'npm run truffle:compile',
  timedOut: false,
  killed: false,
  context: [Object: null prototype] {}
}
```
The immediate reason of this error (as far as I understand) is
```
Error: Cannot find module babel-register
Require stack:
- /home/jahr/ankr/matic-cli/devnet/code/contracts/truffle-config.js
- /home/jahr/.nvm/versions/node/v16.17.1/lib/node_modules/truffle/node_modules/original-require/index.js
- /home/jahr/.nvm/versions/node/v16.17.1/lib/node_modules/truffle/build/cli.bundled.js
```
The full output is here [https://pastebin.com/f6Q5pxgp](https://pastebin.com/f6Q5pxgp)
If I go to the matic contracts folder,  change node version to 8.11.3 as described here [https://github.com/maticnetwork/contracts/issues/383](https://github.com/maticnetwork/contracts/issues/383) and execute `npm i`,  `npm run truffle:compile` succeeds after that. But I can't continue installation or use this node version from very beginning due to other errors.

Npm version output is 
```
$ npm version
{
  '@maticnetwork/matic-cli': '0.0.4',
  npm: '8.15.0',
  node: '16.17.1',
  v8: '9.4.146.26-node.22',
  uv: '1.43.0',
  zlib: '1.2.11',
  brotli: '1.0.9',
  ares: '1.18.1',
  modules: '93',
  nghttp2: '1.47.0',
  napi: '8',
  llhttp: '6.0.9',
  openssl: '1.1.1q+quic',
  cldr: '41.0',
  icu: '71.1',
  tz: '2022a',
  unicode: '14.0',
  ngtcp2: '0.1.0-DEV',
  nghttp3: '0.1.0-DEV'
}
```

Can you help me with this issue? Any help is really appreciated, my node knowledge is very basic.

**Comment by GHORALL** [comment]
@gladcow How long have you been experiencing this error?

**Comment by MatthewDEVi** [comment]
@gladcow are you still facing the issue mate? if yes kindly response now

**Comment by marcello33** [comment]
Can you try with the following contracts branch?
`CONTRACTS_BRANCH=mardizzone/node-16`

**Comment by gladcow** [comment]
@GHORALL ,  I have tried to start local devnet for several days,  still trying.)

@marcello33 ,  thank you,  let me try,  I will write results here in a 1-2 hours.

**Comment by gladcow** [comment]
@marcello33 I have tried fresh matic-cli/master installation. It uses `mardizzone/node-16` as contract branch already (https://github.com/maticnetwork/matic-cli/blob/a1a75dc2595c83160670921bcd1cc8a31d8d0d19/configs/devnet/docker-setup-config.yaml#L10). I've got an error:
```
$ ../bin/matic-cli setup devnet -c ../configs/devnet/docker-setup-config.yaml
✔ Clone Heimdall repository
⠋ Build Heimdall
◼ Create testnet files for Heimdall
rm -rf build
mkdir -p build
go build -ldflags '-X github.com/maticnetwork/heimdall/version.Name=heimdall -X github.com/maticnetwork/heimdall/version.ServerName=heimdalld -X github.com/maticnetwork/heimdall/version.ClientName=heimdallcli -X github.com/maticnetwork/heimdall/version.Version=0.3.0-41-g15a6bc2d -X github.com/maticnetwork/heimdall/version.Commit=1✔ Clone Heimdall repository
⠸ Build Heimdall
◼ Create testnet files for Heimdall
go build -ldflags '-X github.com/maticnetwork/heimdall/version.Name=heimdall -X github.com/maticnetwork/heimdall/version.ServerName=heimdalld -X github.com/maticnetwork/heimdall/version.ClientName=heimdallcli -X github.com/maticnetwork/heimdall/version.Version=0.3.0-41-g15a6bc2d -X github.com/maticnetwork/heimdall/version.Commit=1✔ Clone Heimdall repository
⠏ Build Heimdall
◼ Create testnet files for Heimdall
✔ Clone Heimdall repository
✔ Clone Heimdall repository
✔ Build Heimdall
✖ Command failed: /home/user/matic-cli/devnet/code/heimdall/build/heimdalld create-testnet --home devnet --v 1 --n NaN --chain-id heimdall-4052 --node-host-pr…
node:internal/process/promises:279
            triggerUncaughtException(err, true /* fromPromise */);
            ^

Error: Command failed: /home/user/matic-cli/devnet/code/heimdall/build/heimdalld create-testnet --home devnet --v 1 --n NaN --chain-id heimdall-4052 --node-host-prefix heimdall --output-dir devnet
    at makeError (/home/user/matic-cli/node_modules/execa/index.js:174:9)
    at /home/user/matic-cli/node_modules/execa/index.js:278:16
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async Task.task (/home/user/matic-cli/src/setup/devnet/index.js:701:11)
    at async Task.run (/home/user/matic-cli/node_modules/listr2/dist/index.cjs:979:11)
    at async /home/user/matic-cli/node_modules/listr2/node_modules/p-map/index.js:57:22 {
  code: 1,
  stdout: null,
  stderr: null,
  failed: true,
  signal: null,
  cmd: '/home/user/matic-cli/devnet/code/heimdall/build/heimdalld create-testnet --home devnet --v 1 --n NaN --chain-id heimdall-4052 --node-host-prefix heimdall --output-dir devnet',
  timedOut: false,
  killed: false
}
```

**Comment by marcello33** [comment]
I see `create-testnet --home devnet --v 1 --n NaN` so the number of nonValidators (sentry nodes) is not set correctly.
Can I see your configs file @gladcow ?
Thanks

**Comment by gladcow** [comment]
What is working for me is using `v0.0.9` tag for matic-cli and changing `contractsBranch' in `configs/devnet/docker-setup-config.yaml` to `mardizzone/node-16`. Now it is completed without the errors,  thank you guys,  it was really helpful.)

@marcello33,  about your question about my configs: I didn't edit anything manually,  all configs were used as is,  as in the current `master` branch, commit https://github.com/maticnetwork/matic-cli/commit/a1a75dc2595c83160670921bcd1cc8a31d8d0d19 If you need some additional information from me to investigate the issue, tell me please,  I'll sent it to you, but my problem is resolved with using `v0.0.9` and contract branch you recommended, it's ok for me.

**Comment by marcello33** [comment]
Thanks for getting back! Happy it worked
